src/__/collections/groupBy.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace collections;
+
+/**
+ * Returns an associative array where the keys are values of $key.
+ *
+ * Based on {@author Chauncey McAskill}'s {@link https://gist.github.com/mcaskill/baaee44487653e1afc0d array_group_by()}
```

Comment:
:+1: 

For the licensing part, the work of @mcaskill is based on @jakezatecky, which itself is [licensed under MIT](https://github.com/jakezatecky/array_group_by/blob/master/LICENSE.txt).

@mcaskill does not specify any license, but with the author mention here this seems ok for me. You can send a message to him to make things more explicit!

Reply:
Thanks for the props! Indeed, I've forgotten to add a license to my gists with a composer manifest. This is totally fine with me.